### PR TITLE
Anti Spam 3D + Fix Markers Admin

### PR DESCRIPTION
**Pull request by NiiRoZz**
Test for Anti Spam to 3D :
* CreateVehicle ---> CreateVehicleLocal
Do spam  Ref to nonnetwork object inside RPT server but it's BIS fault
* Change colorvehicle : don't send remoteexec and setvariable when 3D
preview
* Opti with Life_Pos_Attach
* Cleaned up VehicleShopEnd3DPreview

#746
Send me if have Spam with 3D, i will see if can fix it

After @BoGuu Comment and Modified 3D Spam commit : 
* Deleted False inside remoteExecCall of colorvehicle + Modified setObjectTextureGlobal to setObjectTexture when 3D Preview

### CODE REVIEW

File: Altis_Life.Altis/core/vehicle/fn_colorVehicle.sqf [hunk 1]
```diff
@@ -35,8 +37,10 @@ if (!isClass (missionConfigFile >> "LifeCfgVehicles" >> _classNameLife)) then {
...
 _textures = SEL(SEL(M_CONFIG(getArray,"LifeCfgVehicles",_classNameLife,"textures"),_index),2);
-if (isNil "_textures" || {count _textures isEqualTo 0}) exitWith {};
+if (isNil "_textures" || {_textures isEqualTo []}) exitWith {};
 
-_vehicle setVariable ["Life_VEH_color",_index,true];
+if (!_3D) then {
+    _vehicle setVariable ["Life_VEH_color",_index,true];
+};
 
 {_vehicle setObjectTextureGlobal [_forEachIndex,_x];} forEach _textures;
```
Comment (BoGuu) [inline Altis_Life.Altis/core/vehicle/fn_colorVehicle.sqf]: If are aiming to stay local, setObjectTextureGlobal wouldn't be the best command, drop the global, differentiate between the preview like on the others :)

File: life_hc/MySQL/Vehicles/fn_spawnVehicle.sqf [hunk 2]
```diff
@@ -101,7 +101,7 @@ _vehicle allowDamage true;
 [_pid,_side,_vehicle,1] remoteExecCall ["TON_fnc_keyManagement",RSERV];
 _vehicle lock 2;
 //Reskin the vehicle
-[_vehicle,_vInfo select 8] remoteExecCall ["life_fnc_colorVehicle",_unit];
+[_vehicle,_vInfo select 8,false] remoteExecCall ["life_fnc_colorVehicle",_unit];
```
Comment (BoGuu) [inline life_hc/MySQL/Vehicles/fn_spawnVehicle.sqf]: Default is false

**Comment by tkcjesse** [comment]
Your spacing in life_server/functions.sqf is like.... off the walls.

**Comment by NiiRoZz** [comment]
Yeah but i don't love to put 4 spaces because my TAB are 2 spaces

**Comment by Jawshy** [comment]
> Yeah but i don't love to put 4 spaces because my TAB are 2 spaces
 Show all checks

Then change your tabs to equal 4 spaces as per https://github.com/ArmaLife/Framework/issues/646.

**Comment by Jawshy** [comment]
#### Anti Spam 3D 

###### arma3_xxxx-xx-xx_xx-xx-xx.rpt: 
When a land vehicle is deselected or the shop menu/dialog is closed: 
```
17:31:06 Ref to nonnetwork object 77f82b00# 1782464: hatchback_01_f.p3d
```

When an air vehicle is deselected or the shop menu/dialog is closed: 
```
17:46:22 Ref to nonnetwork object b6040100# 1783489: heli_light_01_civil_f.p3d
```

###### createVehicle.log: 
On land vehicle purchase: 
```
26.05.2016 17:31:07: Jawsh (10.0.0.7:2316) 90195302929df64cd18b06dc157c27e4 - #0 "C_Hatchback_01_F" 3:53 [16995,12857,17]
```

When an air vehicle is deselected or the shop menu/dialog is closed: 
```
26.05.2016 17:48:04: Jawsh (10.0.0.7:2316) 90195302929df64cd18b06dc157c27e4 - #0 "B_Parachute" 3:79 [0,0,0]
```

On air vehicle purchase: 
```
26.05.2016 17:50:08: Jawsh (10.0.0.7:2316) 90195302929df64cd18b06dc157c27e4 - #0 "B_Heli_Light_01_F" 3:87 [26868,24567,25]
```

###### remoteExec.log: 
On land vehicle purchase: 
```
26.05.2016 17:31:07: Jawsh (10.0.0.7:2316) 90195302929df64cd18b06dc157c27e4 - #0 "ton_fnc_setobjvar [<NULL-object>,"color",1,true]"
26.05.2016 17:31:07: Jawsh (10.0.0.7:2316) 90195302929df64cd18b06dc157c27e4 - #0 "ton_fnc_setobjvar [22188100# 1815089: hatchback_01_f.p3d REMOTE,"trunk_in_use",false,true]"
26.05.2016 17:31:07: Jawsh (10.0.0.7:2316) 90195302929df64cd18b06dc157c27e4 - #0 "ton_fnc_setobjvar [22188100# 1815089: hatchback_01_f.p3d REMOTE,"vehicle_info_owners",[["76561198025395979","Jawsh"]],true]"
26.05.2016 17:31:07: Jawsh (10.0.0.7:2316) 90195302929df64cd18b06dc157c27e4 - #0 "ton_fnc_keymanagement ["76561198025395979",CIV,22188100# 1815089: hatchback_01_f.p3d REMOTE,1]"
26.05.2016 17:31:07: Jawsh (10.0.0.7:2316) 90195302929df64cd18b06dc157c27e4 - #0 "ton_fnc_vehiclecreate ["76561198025395979",CIV,22188100# 1815089: hatchback_01_f.p3d REMOTE,5]"
26.05.2016 17:31:07: Jawsh (10.0.0.7:2316) 90195302929df64cd18b06dc157c27e4 - #0 "db_fnc_updatepartial ["76561198025395979",CIV,438237,0]"
```

On air vehicle purchase: 
```
26.05.2016 17:50:08: Jawsh (10.0.0.7:2316) 90195302929df64cd18b06dc157c27e4 - #0 "ton_fnc_setobjvar [26941600# 1815126: heli_light_01_f.p3d REMOTE,"trunk_in_use",false,true]"
26.05.2016 17:50:08: Jawsh (10.0.0.7:2316) 90195302929df64cd18b06dc157c27e4 - #0 "ton_fnc_setobjvar [26941600# 1815126: heli_light_01_f.p3d REMOTE,"vehicle_info_owners",[["76561198025395979","Jawsh"]],true]"
26.05.2016 17:50:08: Jawsh (10.0.0.7:2316) 90195302929df64cd18b06dc157c27e4 - #0 "ton_fnc_keymanagement ["76561198025395979",CIV,26941600# 1815126: heli_light_01_f.p3d REMOTE,1]"
26.05.2016 17:50:08: Jawsh (10.0.0.7:2316) 90195302929df64cd18b06dc157c27e4 - #0 "ton_fnc_vehiclecreate ["76561198025395979",CIV,26941600# 1815126: heli_light_01_f.p3d REMOTE,9]"
26.05.2016 17:50:08: Jawsh (10.0.0.7:2316) 90195302929df64cd18b06dc157c27e4 - #0 "db_fnc_updatepartial ["76561198025395979",CIV,942227,0]"
```

###### setVariable.log:
On land vehicle purchase: 
```
26.05.2016 17:31:07: Jawsh (10.0.0.7:2316) 90195302929df64cd18b06dc157c27e4 - #0 "life_veh_color" = 5 3:53 C_Hatchback_01_F
```

On air vehicle purchase: 
```
26.05.2016 17:50:08: Jawsh (10.0.0.7:2316) 90195302929df64cd18b06dc157c27e4 - #0 "life_veh_color" = 9 3:87 NetworkMessageCreateHelicopterRTD
```

###### addBackpackCargo.log: 
On air vehicle purchase: 
```
26.05.2016 17:50:08: Jawsh (10.0.0.7:2316) 90195302929df64cd18b06dc157c27e4 - #0 3:93 B_Parachute 0:0
```

#### Fix Markers Admin 

Tested with no issues found. 

Ref: https://github.com/ArmaLife/Framework/issues/365

**Comment by BoGuu** [comment]
We can fiddle around with the BEfilters to reduce the logging in the .logs right @Jawshy ? As for the rpt log, like I said before, it's client, and it's a BI problem, we shouldn't be syncing all our vehicle shop preview changing with the whole server, so the slight RPT spam is a compromise. 

Also @NiiRoZz, I see you've replaced the for loops with while loops. However, to achieve what you are trying to, you can leave the for loops in and change

`if (!life_markers && !(alive _PlayersOrVehicles) && (vehicle _PlayersOrVehicles != _PlayersOrVehicles) && (getPlayerUID _PlayersOrVehicles != """")) exitWith {};`

Into

`if (!life_markers || !(alive _PlayersOrVehicles) || (vehicle _PlayersOrVehicles != _PlayersOrVehicles) || (getPlayerUID _PlayersOrVehicles != """")) exitWith {};`

But this all this marker stuff needs to be rewritten

**Comment by Jawshy** [comment]
> We can fiddle around with the BEfilters to reduce the logging in the .logs right ?

@BoGuu, we could, but I don't think it's going to be necessary as most of the logs are only written on vehicle purchase which is significantly harder to spam than switching between vehicles in the menu. 

The only BE log that can be easily spammed by players is the createVehicle.log with parachutes, however server admins can specify a maxCreateVehiclePerInterval command (see [BattlEye Documentation](https://www.battleye.com/support/documentation/)) in their BEServer.cfg file which will kick players that excessively spam createVehicle. 

> As for the rpt log, like I said before, it's client, and it's a BI problem, we shouldn't be syncing all our vehicle shop preview changing with the whole server, so the slight RPT spam is a compromise.

Indeed. The client/local.rpt spam is not really an issue as that would only affect the player and no one else.

**Comment by BoGuu** [comment]
Did you see my message above NiiRoZz? This ideally needs to be re written entirely as well, it's a train wreck.

**Pull request closed without merging**